verification/client.go

```diff
@@ -33,14 +33,24 @@ type DPETCI struct {
 	CurrentTCI    []byte
 }
 
+type DPESignedHash struct {
+	Handle           ContextHandle
+	HmacOrSignatureR []byte
+	SignatureS       []byte
+}
+
 type DPEClient interface {
 	InitializeContext(flags InitCtxFlags) (*ContextHandle, error)
 	GetProfile() (*GetProfileResp, error)
+	DeriveChild(handle *ContextHandle, inputData []byte, flags DeriveChildFlags, tciType uint32, targetLocality uint32) (*DeriveChildResp, error)
 	CertifyKey(handle *ContextHandle, label []byte, format CertifyKeyFormat, flags CertifyKeyFlags) (*CertifiedKey, error)
 	GetCertificateChain() ([]byte, error)
 	TagTCI(handle *ContextHandle, tag TCITag) (*ContextHandle, error)
 	GetTaggedTCI(tag TCITag) (*DPETCI, error)
+	ExtendTCI(handle *ContextHandle, inputData []byte) (*ContextHandle, error)
 	DestroyContext(handle *ContextHandle, flags DestroyCtxFlags) error
+	RotateContextHandle(handle *ContextHandle, flags RotateContextHandleFlags) (*ContextHandle, error)
+	Sign(handle *ContextHandle, label []byte, flags SignFlags, toBeSigned []byte) (*DPESignedHash, error)
 }
```

Comment:
Could you add all the client code in one separate PR without the tests? All of the other PRs you've sent depend on it, so I think it will make all of the PRs quite a bit smaller.

Reply:
sure I shall raise one

Reply:
Please find PR link here: https://github.com/chipsalliance/caliptra-dpe/pull/255